chaste_codegen/chaste_model.py

```diff
@@ -350,103 +318,151 @@ def _get_modifiable_parameters(self):
 
         Stimulus currents are ignored and the result is sorted by display name"""
         tagged = self._model.get_variables_by_rdf((self._PYCMLMETA, 'modifiable-parameter'), 'yes')
+        # Get derived quantities excluding stimulus current params
         annotated = [q for q in self._model.variables()
-                     if self._model.has_ontology_annotation(q, self._OXMETA)]
-        currents = [var for var in annotated if self._model.get_ontology_terms_by_variable(var, self._OXMETA)[-1]
-                    .startswith('membrane_stimulus_current')]
+                     if self._model.has_ontology_annotation(q, self._OXMETA)
+                     and q not in self._stimulus_params]
 
         parameters = set(tagged + annotated) -\
-            set(currents + self._model.get_derived_quantities() + self._state_vars + [self._time_variable])
+            set(self._model.get_derived_quantities() + self._state_vars + [self._time_variable])
         return sorted(parameters, key=lambda v: self._model.get_display_name(v, self._OXMETA))
 
     def _get_membrane_stimulus_current(self):
-        """ Find the membrane_stimulus_current variable if it exists"""
-        try:
+        try:  # get membrane_stimulus_current
             return self._model.get_variable_by_ontology_term((self._OXMETA, 'membrane_stimulus_current'))
         except KeyError:
             self._logger.info(self._model.name + ' has no membrane_stimulus_current')
             self._is_self_excitatory = len(list(self._model.get_rdf_annotations(subject=self._model.rdf_identity,
                                                 predicate=(self._PYCMLMETA, 'is-self-excitatory'), object_='yes'))) > 0
             return None
 
+    def _convert_membrane_stimulus_current(self):
+        """ Find the membrane_stimulus_current and convert it to uA_per_cm2, using GetIntracellularAreaStimulus"""
+        if self._membrane_stimulus_current_orig is None:
+            return None, self._stimulus_units
+        # get derivative equations
+        d_eqs = self.get_equations_for(self._y_derivatives)
```

Comment:
Do we want to be stripping units here? I think we do because we evaluate below.

(Incidentally, is there a case for the default changing?)

Reply:
yes we do and know please don't change the deafult, it would be a pain to have to put strip_units=true everywhere plus the default result is probably closer to what we expect (an expression we can work with without strange variable wrapped units)